edustripetech/auth_service_v1
# Setup caching with Redis

#### What does this PR do?
- This PR enables caching on the platform to avoid unnecessary "expensive" calls.

#### Description of Task proposed in this pull request?
- Install dependencies and create cache function
- Implement caching on home route

#### How should this be manually tested (Quality Assurance)?
- Firstly ensure you have the Redis server [installed and running](https://redis.io/topics/quickstart) locally on your machine.
- Start up the development server.
- Make a request to the home route `GET /api/v1/` taking note of the time of the request using a tool like Postman.
- Make another request to the same endpoint and note the time difference as well.

#### What are the relevant pivotal tracker stories?
- [#172622995](https://www.pivotaltracker.com/story/show/172622995)

#### Any background context you want to add (Operations Impact)?
- [Install and start up](https://redis.io/topics/quickstart) the Redis server locally on your machine.
- Create a new env variable `REDIS_PORT`
- This is just the basic implementation. The team will need to discuss caching strategies. 

#### Screenshots:
-- First request
![Screenshot (154)](https://user-images.githubusercontent.com/44305641/82143687-9b0d7980-983d-11ea-8f71-a4756db012cc.png)

-- Second (cached) request
![Screenshot (155)](https://user-images.githubusercontent.com/44305641/82143695-a6f93b80-983d-11ea-8413-4401ffbbc5d6.png)
